commands/env.go

```diff
@@ -182,6 +186,10 @@ func shellCfgUnset(c CommandLine, api libmachine.API) (*ShellConfig, error) {
 		shellCfg.Prefix = "(setenv \""
 		shellCfg.Suffix = ")\n"
 		shellCfg.Delimiter = "\" nil"
+	case "tcsh":
+		shellCfg.Prefix = "setenv "
```

Comment:
You're right, I think unsetenv would be better.  I'll try that when I have a moment.